lib/types/text.js

```diff
@@ -51,7 +52,7 @@ function text (options) {
     : type
 
   function parse (buf) {
-    return buf
+    return parser(buf)
```

Comment:
We probably don't need two levels of functions here. Just replace parse with a variable.